src/test/java/htsjdk/samtools/CRAMFileReaderTest.java

```diff
@@ -240,4 +240,35 @@ public void testCramIteratorWithoutCallingHasNextFirst() throws IOException {
         Assert.assertNotNull(iterator.next());
         Assert.assertThrows(NoSuchElementException.class, iterator::next);
     }
+
+    // confirm that we can apply a SAMFileSource to SAMRecords via enableFileSource()
+
+    @Test
+    public void enableFileSourceTest() throws IOException {
+        // need to use the real reference here because the iterator validates it
+        final ReferenceSource refSource = new ReferenceSource(REFERENCE_FILE);
+
+        try (final FileInputStream fileStream = new FileInputStream(CRAM_WITH_CRAI);
+             final SamReader samReader = SamReaderFactory
+                .make()
+                .referenceSource(refSource)
+                .open(CRAM_WITH_CRAI)) {
+
+            // note: CRAMFileReader.getIterator() replaces its underlying CRAMIterator
+            // if CRAMFileReader was constructed using a File argument.
+            // here we use the InputStream constructor instead to avoid this
+
+            final CRAMFileReader cramReader = new CRAMFileReader(fileStream, INDEX_FILE, refSource, ValidationStringency.STRICT);
+
+            // TODO: enableFileSource() only affects the *current* underlying CRAMIterator
+            // which may get replaced.  is this correct behavior?
+
+            cramReader.enableFileSource(samReader, true);
+
+            final SAMRecord record = cramReader.getIterator().next();
+            Assert.assertEquals(record.getFileSource().getReader(), samReader);
+
+            cramReader.close();
+        }
+    }
```

Comment:
Yeah, it looks like you had to really jump through hoops to get that code block to execute. But I see the problem now - the SamReader is only propagated by `CRAMFileReader` to the `CRAMIterator` thats in place when enableFileSource is called, which is almost always immediately replaced by a new iterator instance as soon as an iterator is called for.

Given that its so hard to even get this code block to execute, and how unnatural the test code is (requiring opening the same file twice), I'd say we should just remove this test; I created https://github.com/samtools/htsjdk/issues/1293 to deal with the bigger issue.